repos/c2sm/packages/icon/package.py

```diff
@@ -291,6 +313,11 @@ def setup_build_environment(self, env):
             for d in link_dirs:
                 env.append_path('SPACK_COMPILER_IMPLICIT_RPATHS', d)
 
+        if 'none' not in self.spec.variants['dsl'].value:
+            env.set("CUDAARCHS", self.spec.variants['gpu'].value)
+            env.unset("CUDAHOSTCXX")
```

Comment:
Why do you modfify these env-vars?
Are they used in CMAKE?
And what set CUDAHOSTCXX in the first place?

Reply:
Most likely it's cuda, who sets CUDAHOSTCXX: https://github.com/spack/spack/blob/ddecf070454c3440c3342ab92f067b71c52a8336/var/spack/repos/builtin/packages/cuda/package.py#L512